commit 8960318bdde32ba8ea38e268fea8ba130b2ae3ba
Author: Keno Fischer <keno@juliacomputing.com>
Date:   Mon Oct 5 03:28:56 2020 -0400

    Drop AC_FUNC_*ALLOC configure macros
    
    These macros are supposed to replace all uses of alloc/malloc/realloc
    by rpl_* of these versions, where ngspice itself is supposed to define
    the repl_* versions to correct the behavior of `malloc(0)`. However,
    ngspice does not provide these replacement functions, and moreover
    the autoconf check is quite brittle and easily breaks even on
    confirming libc versions and even more likely in cross-compile
    settings. Unsurprisingly, there are several ngspice compilation
    guides on the internet, that simply pass ac_cv_func_malloc_0_nonnull=yes
    to configure, which essentially disables these checks. Given that
    ngspice does not ship the rpl_* versions of these functions, I
    don't believe these checks ever did anything other than break
    people's builds. Further, non-conforming alloc functions are
    much less of a concern these days, so it's unlikely to be
    necessary to perform such a replacement. Dropping these macros
    is consistent with what I have seen other projects do, e.g. [1]
    
    [1] https://patches.linaro.org/patch/110647/

diff --git a/configure.ac b/configure.ac
index 2419e69f1..dae84f91e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -637,9 +637,6 @@ AC_HEADER_STDBOOL
 AC_TYPE_SIZE_T
 
 # Checks for library functions.
-AC_FUNC_ALLOCA
-AC_FUNC_MALLOC
-AC_FUNC_REALLOC
 AC_FUNC_CLOSEDIR_VOID
 AC_FUNC_SELECT_ARGTYPES
 AC_FUNC_SETVBUF_REVERSED
